From 8d8631c4b507204bd0ec0642ca35386c83c81d5f Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 26 May 2014 11:49:35 -0400
Subject: [PATCH] Store EntityMetaData by EntityId

Currently unloading then reloading an Entity is making the TempMetaData persist across these transitions.

This makes hard to use TempMeta to know if something needs to be initialized or not (listeners) and is carrying
over old listeners to the old entity that no longer exists.

Using the EntityId + world name is guaranteed unique per server session and therefor will be unique to the living entity's object.
---
 src/main/java/org/bukkit/craftbukkit/metadata/EntityMetadataStore.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/bukkit/craftbukkit/metadata/EntityMetadataStore.java b/src/main/java/org/bukkit/craftbukkit/metadata/EntityMetadataStore.java
index 35c484f..d89b17d 100644
--- a/src/main/java/org/bukkit/craftbukkit/metadata/EntityMetadataStore.java
+++ b/src/main/java/org/bukkit/craftbukkit/metadata/EntityMetadataStore.java
@@ -18,6 +18,6 @@ public class EntityMetadataStore extends MetadataStoreBase<Entity> implements Me
      */
     @Override
     protected String disambiguate(Entity entity, String metadataKey) {
-        return entity.getUniqueId().toString() + ":" + metadataKey;
+        return entity.getWorld().toString() + ":" + entity.getEntityId() + ":" + metadataKey;
     }
 }
-- 
1.9.1

